docs/release-procedure.md

```diff
@@ -23,29 +24,49 @@ as reviewing them again would not add a lot of value but would add lots of exces
+### Preparing the repo for a Major or Minor Version
 
-**Note**: from now on you work in a branch in your fork (in `origin`).
+**Note**: this should be done in your fork of Modin.
 
-Change the `README.md` in the repo root so release badge points to new (would be created) release.
+First, update your fork of Modin's master with the main repo's master. From your master, create a new
+branch called `release-X.Y.0` off of master. Create an empty commit in your new branch with the message
+`Release version X.Y.0`. Make sure to sign this commit with both your GPG key
+and with the conventional `git commit -s` (so `git commit -s -S`). Open a PR against modin-project/modin with just this commit.
 
-### Commit
+### Preparing the repo for a Patch Version
 
-        git commit -a -S -m "Release version X.Y.Z"
+**Note**: this should be done in the original Modin repository (in `upstream`) .
 
-Pay attention to that `-S` switch if you haven't enabled signing all the commits!
-Modin requires release commits to be signed (do not mix this with "signing off" commits, this
-is a whole different story - signed commits are cryptographically signed).
+First, you must create a new branch in the upstream (main modin-project/modin) repo for the new release.
+This branch must be named `release-X.Y.Z`, and should be made off of the tag for the last release. To
+do this, look up the commit that corresponds to the last release, copy its SHA, and then do 
+`git checkout <PREVIOUS_RELEASE_COMMIT_SHA>`. Next, do `git checkout -b release-X.Y.Z` to create
+the branch for the new release, and push it upstream.
 
-Push the commit, make a PR _against `release-X.Y.Z`_ branch and, when it's merged, pull that branch from `upstream` before proceeding.
+**Note**: now you must switch to your fork of Modin.
+
+From your fork of Modin, fetch the upstream repo, and checkout the release branch you made above.
+From this release branch, create a new branch, and cherry-pick the commits that will go into this release.
+
+Once these commits have been applied to your branch, edit the `README.md` so that the PyPi badge will
+point to the badge for this specific version (instead of latest) and so that the docs link will point
+to the docs for this specific version (rather than latest).
+
+Once this has been committed, create an empty commit, the same as for a major or minor version,
+with the message `Release version X.Y.Z`, and make sure to sign it with both your GPG key, and the
+traditional git sign-off. Create a PR using your branch against the `release-X.Y.Z` branch in the
+original Modin repo.
 
 ### Tag commit
 
-**Note**: from now on you work in the release branch (in `upstream`).
+After the PR has been merged, clone a clean copy of the Modin repo from the modin-project organization.
+You now need to tag the commit that corresponds to the above PR with the appropriate tag for this release.
+
+**Note**: from now on you work on the master branch (in `upstream`) .
```

Comment:
this feels wrong (I mean "master branch"), as we should be working on `release-X.Y.Z` branch checked from `upstream` here

Reply:
It has to be `master` if its a major or minor release, since the commit we tag goes into master. I'll amend it to be the release branch if its a patch version.